### DISCUSSION TITLE
Closing the application between tests

### DISCUSSION THREAD

**Discussion by sqbeats** [appium/appium#19766] [Q&A]
How can I make sure that my application does not close when the tests in the script are finished? For example, I have 2 scripts with tests, when the tests in the first script finish, the tests from the second begin to be checked.

**Comment by jlipps** [comment]
It's best to have one test script per session. But if you want to run multiple tests within the same session, you should probably have Appium session creation and deletion in your global suite setup and teardown, not your per test setup and teardown.

**Reply by sqbeats** [reply]
I tried use pytest.fixture(scope='session'), but it didn't work correctly

**Reply by sqbeats** [reply]
I solved the problem by reworking the test logic
